data/systemd/snapd.core-fixup.sh

```diff
@@ -11,6 +11,38 @@ if ! grep -q "ID=ubuntu-core" /etc/os-release; then
     exit 0
 fi
 
+# Workaround https://forum.snapcraft.io/t/5253
+#
+# We see sometimes corrupted uboot.env files created by fsck.vfat.
+# On the fat filesystem they are indistinguishable because one
+# has a fat16 name UBOOT.ENV (and not lfn (long-file-name)) but
+# the other has a "uboot.env" lfn name and a FSCK0000.000 FAT16
+# name. The only known workaround is to remove all dupes and put
+# one file back in place.
+if [ $(ls /boot/uboot | grep ^uboot.env$ | wc -l) -gt 1 ]; then
+    echo "Corrupted uboot.env file detected"
+    # ensure we have one uboot.env to go back to
+    cp -a /boot/uboot/uboot.env /boot/uboot/uboot.env.save
```

Comment:
What I'm worried about here if the uboot.env file we pick up is deterministic and if not, does that matter. Can you add a comment please.